tox.ini

```diff
@@ -3,17 +3,49 @@
 ; installed this repository in 'developer mode'!
 
 [tox]
+passenv = PYTHONPATH
+skip_install = True
+skip_missing_interpreters=true
 skipsdist = True
 envlist =
-    docs,
-    pep8,
+    py,
+    linkcheck,
+    style,
 
-[testenv:pep8]
-passenv = PYTHONPATH
-deps =
-    -rrequirements-dev.txt
+[testenv]
+whitelist_externals =
+    pytest
+commands =
+    pytest \
+        --cov-fail-under=55 \
+        --cov-report=html \
+        --cov-report=xml \
+        --cov=tensorwaves
+
+; Allows to quickly build html with `tox -e doc` from the root dir
+[testenv:doc]
+setenv =
+    NBSPHINX_EXECUTE = "yes"
+changedir = doc
+whitelist_externals =
+    make
+commands =
+    make html
+
+[testenv:linkcheck]
+setenv =
+    NBSPHINX_EXECUTE = "yes"
+changedir = doc
+whitelist_externals =
+    make
+commands =
+    make linkcheck
+
+[testenv:style]
+whitelist_externals =
+    pre-commit
 commands =
-    flake8
+    pre-commit run -a
```

Comment:
Running style checks means running all of pre-commit?

Reply:
Yeah I start to see pre-commit as a collection of style checks (at least true if you look at the hooks that we currently implemented)

Reply:
but it also runs mypy right?

Reply:
Right, and I guess technically mypy is not a style check. But the `testenv` section names of a `tox.ini` file have a practical usage. As in, you want to keep them short and easy to type (that's why the command is the super simple '`tox`'), so that you can for instance do:

```bash
tox -e py    # run pytest
tox -e style # run 'style' checks
tox -e doc   # test-build documentation
```